Realm/ObjectStore/object_store_exceptions.hpp

```diff
@@ -24,42 +24,78 @@
 #include <string>
 
 namespace realm {
+    class Property;
 
     class ObjectStoreException : public std::exception {
     public:
         enum class Kind {
```

Comment:
I'm wondering what the rationale is behind using this approach rather than defining separate exception error types deriving from a common `ObjectStoreException`, which could then also just define extra information as members instead of going through a (potentially expensive) `std::map`. Especially considering that these exceptions will have to be translated into Java exceptions at some point, where it just becomes overhead. :)